src/main/java/seedu/duke/commands/CommandCreator.java

```diff
@@ -36,13 +37,44 @@ public static Command createAddCommand(String commandString, String description,
         }
     }
 
+    /**
+     * Creates and returns an AddRecurringCommand with given arguments.
+     *
+     * @param description description of the task
+     * @param argumentsMap Hashmap containing compulsory and optional arguments
+     * @return AddRecurringCommand with given arguments.
+     * @throws DukeException When description is empty.
+     */
     public static Command createAddRecurringCommand(String description, HashMap<String, String> argumentsMap)
             throws DukeException {
         if (description.equals("")) {
             throw new DukeException(Messages.EXCEPTION_EMPTY_DESCRIPTION);
         }
         return new AddRecurringCommand(description, argumentsMap);
     }
+    
+    public static Command createSpendCommand(String description, HashMap<String, String> argumentsMap)
+            throws DukeException {
+        if (description.equals("")) {
+            throw new DukeException(Messages.EXCEPTION_EMPTY_DESCRIPTION);
+        }
+        if (argumentsMap.containsKey("v")) {
+            if (argumentsMap.get("v").equals("")) {
+                throw new DukeException(Messages.EXCEPTION_EMPTY_VALUE);
+            }
+        }
+        if (argumentsMap.containsKey("currency")) {
+            if (argumentsMap.get("currency").equals("")) {
+                throw new DukeException(Messages.EXCEPTION_EMPTY_CURRENCY);
+            }
+        }
+        if (argumentsMap.containsKey("date")) {
+            if (argumentsMap.get("date").equals("")) {
+                throw new DukeException(Messages.EXCEPTION_EMPTY_DATE);
+            }
+        }
```

Comment:
Maybe putting these 3 checks in SpendCommand's constructor or execute method would be better, since CommandCreator deals with creating Commands. Checking argumentsMap would be done by the Command itself.

Reply:
Sure! Yes it makes sense